### PR TITLE
bakery: remove redundant Storage type

### PR DESCRIPTION
**Pull request by rogpeppe**
The newer storage type provides a superset of the functionality
and is much more versatile. We can remove it because with
the new version we're now free to break API compatibility.

### CODE REVIEW

File: bakery/example/idservice/idservice.go [hunk 1]
```diff
@@ -80,12 +80,13 @@ func (h *handler) userHandler(p httprequest.Params) (interface{}, error) {
 		// Theoretically, we could just redirect the user to the
 		// login page, but that would p.Requestuire a different flow
 		// and it's not clear that it would be an advantage.
-		m, err := h.svc.NewMacaroon(nil, nil, []checkers.Caveat{{
+		m, err := h.svc.NewMacaroon([]checkers.Caveat{{
```
Comment (urosj) [inline bakery/example/idservice/idservice.go:83]: Nice and clean!

File: bakery/mgostorage/rootkey.go [hunk 2]
```diff
@@ -208,14 +208,14 @@ func (s *RootKeys) setCurrent(policy Policy, key rootKey) {
 	s.current[policy] = key
 }
 
-type rootKeyStorage struct {
+type storage struct {
```
Comment (urosj) [inline bakery/mgostorage/rootkey.go:211]: Not sure on this. Storage seems too general.

File: bakery/service.go [hunk 3]
```diff
@@ -143,21 +126,12 @@ func (svc *Service) Check(ms macaroon.Slice, checker FirstPartyChecker) error {
...
 	}
 	id := ms[0].Id()
-	if svc.rkStore != nil {
-		// We're using a RootKeyStore - hack off the
-		// uuid at the end of the id.
-		if i := bytes.LastIndexByte(id, '-'); i >= 0 {
-			id = id[0:i]
-		}
+	// Trim the UUID at the end of the id.
+	if i := bytes.LastIndexByte(id, '-'); i >= 0 {
```
Comment (urosj) [inline bakery/service.go:133]: Please describe why this is needed.

File: bakery/storage.go [hunk 4]
```diff
@@ -1,36 +1,12 @@
...
 	"sync"
 )
 
-// Storage defines storage for macaroons.
-// Calling its methods concurrently is allowed.
-//
-// This type is the original storage interface supported
-// by the bakery and will eventually be deprecated
-// in favour of RootKeyStorage.
+// Storage defines defines storage for macaroon root keys.
```
Comment (urosj) [inline bakery/storage.go]: double defines.

**Comment by mhilton** [comment]
:+1:

**Comment by urosj** [comment]
LGTM, nice clean-up.

File: bakery/mgostorage/rootkey.go (hunk 2 above)
Comment (rogpeppe) [inline bakery/mgostorage/rootkey.go:211]: Note that it's bakery.Storage, not just "storage". It is the only storage that the bakery code writes to, and the old storage type was also just named "storage".

File: bakery/service.go (hunk 3 above)
Comment (rogpeppe) [inline bakery/service.go:133]: done

File: bakery/storage.go (hunk 4 above)
Comment (rogpeppe) [inline bakery/storage.go]: done

**Pull request merged**